HB-06

Max-Pochepets opened the pull request:
One to One

- Models implemented: Shopping cart, Ticket
- DAO's created: Shopping cart, Ticket
- Service created: Shopping cart

Max-Pochepets committed:
first

Max-Pochepets committed:
first

Max-Pochepets committed:
beta || before testing

Max-Pochepets committed:
beta || before testing

Max-Pochepets committed:
beta || tests failed

Max-Pochepets committed:
HB-06 rebase on HB-05

Max-Pochepets committed:
Initial:

task complete

Max-Pochepets committed:
Initial:

task complete

Max-Pochepets committed:
Refactor:

@GeneratedValue annotation removed from ShoppingCart.id

Max-Pochepets committed:
Refactor:

@GeneratedValue annotation removed from ShoppingCart.id

Max-Pochepets committed:
Merge remote-tracking branch 'origin/HB-06' into HB-06

# Conflicts:
#	src/main/java/cinema/shop/App.java

Max-Pochepets committed:
Refactor:

merge changed to save in ShoppingCartDaoImpl.add

vicire commented on src/main/java/cinema/shop/model/ShoppingCart.java:
```diff
@@ -0,0 +1,77 @@
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne(cascade = CascadeType.MERGE)
```
Optional - why do you use cascade here?

Max-Pochepets replied:
I didn't correct this from the time I was using `merge()` on shopping cart. I'll remove it, THX :)

KseniiaMakarova commented on src/main/java/cinema/shop/model/MovieSession.java:
```diff
@@ -17,12 +15,10 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "movie_id")
```
there is no need to remove `@JoinColumn` annotations)

KseniiaMakarova commented on src/main/java/cinema/shop/model/ShoppingCart.java:
```diff
@@ -0,0 +1,77 @@
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne(cascade = CascadeType.MERGE)
```
let's remove cascade here

KseniiaMakarova commented on src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
```
if there is no shopping cart for the user - we still save ticket to the db, is it ok?

KseniiaMakarova commented on src/main/java/cinema/shop/service/impl/ShoppingCartServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ShoppingCart shoppingCartByUser = getByUser(shoppingCart.getUser());
```
do you need this?

Max-Pochepets committed:
Fix:

@JoinColumn in MovieSession.java added
addSession() logic improved
redundant variable in clear() removed
cascade in ShoppingCart.java removed
